unit/solvers/smt2_incremental/smt_to_smt2_string.cpp

```diff
@@ -29,6 +30,15 @@ TEST_CASE(
   CHECK(smt_to_smt2_string(smt_bit_vector_constant_termt{0, 8}) == "(_ bv0 8)");
 }
 
+TEST_CASE(
+  "Test smt_bit_vector extract to string conversion",
+  "[core][smt2_incremental]")
+{
+  CHECK(
+    smt_to_smt2_string(smt_bit_vector_theoryt::extract(7, 3)(
+      smt_bit_vector_constant_termt{0, 8})) == "((_ |extract| 7 3) (_ bv0 8))");
```

Comment:
In the theory, it seems like `extract` has this form:

```((_ extract i j) (_ BitVec m) (_ BitVec n))`

I have two questions:

* Do we need to pass in the second vector as well somehow?
* Is the fact that we quote `extract` inside the `|` going to affect this at all?

Reply:
> In the theory, it seems like `extract` has this form:
> 
> `((_ extract i j) (_ BitVec m) (_ BitVec n))`
> 
> I have two questions:
> 
> * Do we need to pass in the second vector as well somehow?

`extract` has a single parameter `(_ BitVec m)`. `(_ BitVec n)` is the return sort. Compare for example the two parameter unsigned less than operation. This takes two bit vectors and returns a `Bool` sorted result. So it is declared as `(bvult (_ BitVec m) (_ BitVec m) Bool)`
> * Is the fact that we quote `extract` inside the `|` going to affect this at all?

Maybe. This printing implementation keeps the printed form consistent between indexed and non-indexed identifiers. We already know this causes issues for CVC4, I think the appropriate fix will be to quote only where needed. However I'd like to keep this fixing work in a separate ticket / PR and maintain the status quo for this PR.